tests/samplers_tests/test_nsgaii.py

```diff
@@ -146,6 +147,78 @@ def constraints_func(trial: FrozenTrial) -> Sequence[float]:
         assert trial.system_attrs[_CONSTRAINTS_KEY] == (trial.number,)
 
 
+def test_constrained_dominates() -> None:
+
+    directions = [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE]
+
+    def check_hasse_diagram(
```

Comment:
I do not prefer to include this type of logic in the test. Is it difficult to enumerate the test of the relationship between two trials, as in the `_dominates` test?

Reply:
I wanted to check the transitivity of `_constrained_dominates`. If I write down all the relationship, it would need `2 * 11 ** 2` comparisons.

Reply:
Thanks for the clarification. I understood that this test tested many things in a concise manner. Sounds great. Now, we can accept this logic in the test of `_constrained_dominates`, and I think it can be used in the `_dominates`. I would like to hear other opinion. What do you think? > @toshihikoyanase

Reply:
Hmm, I'm not checked in details, but it may be difficult to find which test cases this test function covers. For example, the trial with the constraint of `-inf, inf` is missing, but I'm not sure if it is intentional. IMHO, we can divide this function in terms of the test scenario.